coverage_point_calculator/src/lib.rs

```diff
@@ -0,0 +1,933 @@
+        };
+
+        Ok(Self {
+            radio_type,
+            speedtests: Speedtests::new(speedtests),
+            location_trust_scores,
+            radio_threshold,
+            covered_hexes,
+            boosted_hex_eligibility: boosted_hex_status,
+        })
+    }
+}
+
+impl CoveragePoints {
+    pub fn iter_boosted_hexes(&self) -> impl Iterator<Item = CoveredHex> {
+        let eligible = self.boosted_hex_eligibility.is_eligible();
+
+        self.covered_hexes
+            .clone()
+            .into_iter()
+            .filter(move |_| eligible)
+            .filter(|hex| hex.boosted_multiplier.is_some())
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum BoostedHexStatus {
+    Eligible,
+    WifiLocationScoreBelowThreshold(Decimal),
+    RadioThresholdNotMet,
+}
+
+impl BoostedHexStatus {
+    fn new(
+        radio_type: &RadioType,
+        location_trust_score: Decimal,
+        radio_threshold: &RadioThreshold,
+    ) -> Self {
+        // hip93: if radio is wifi & location_trust score multiplier < 0.75, no boosting
+        if radio_type.is_wifi() && location_trust_score < dec!(0.75) {
+            return Self::WifiLocationScoreBelowThreshold(location_trust_score);
+        }
+
+        // hip84: if radio has not met minimum data and subscriber thresholds, no boosting
+        if !radio_threshold.threshold_met() {
+            return Self::RadioThresholdNotMet;
+        }
+
+        Self::Eligible
+    }
+
+    fn is_eligible(&self) -> bool {
+        matches!(self, Self::Eligible)
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum RadioType {
+    IndoorWifi,
+    OutdoorWifi,
+    IndoorCbrs,
+    OutdoorCbrs,
+}
+
+impl RadioType {
+    fn base_coverage_points(&self, signal_level: &SignalLevel) -> Result<Decimal> {
+        let mult = match self {
+            RadioType::IndoorWifi => match signal_level {
+                SignalLevel::High => dec!(400),
+                SignalLevel::Low => dec!(100),
+                other => return Err(Error::InvalidSignalLevel(*other, *self)),
+            },
+            RadioType::OutdoorWifi => match signal_level {
+                SignalLevel::High => dec!(16),
+                SignalLevel::Medium => dec!(8),
+                SignalLevel::Low => dec!(4),
+                SignalLevel::None => dec!(0),
+            },
+            RadioType::IndoorCbrs => match signal_level {
+                SignalLevel::High => dec!(100),
+                SignalLevel::Low => dec!(25),
+                other => return Err(Error::InvalidSignalLevel(*other, *self)),
+            },
+            RadioType::OutdoorCbrs => match signal_level {
+                SignalLevel::High => dec!(4),
+                SignalLevel::Medium => dec!(2),
+                SignalLevel::Low => dec!(1),
+                SignalLevel::None => dec!(0),
+            },
+        };
+        Ok(mult)
+    }
+
+    fn rank_multipliers(&self) -> Vec<Decimal> {
```

Comment:
this doesn't need to return an owned vec. Instead, make these constants and return a static slice:
```suggestion
    fn rank_multipliers(&self) -> &'static [Decimal] {
```

Reply:
But also, I'm a little wary of using the rank to index a vec. Instead, I would do a match around the rank and radio type

Reply:
Ended up going with a match statement, it reads clearly and avoids allocating a structure to indexing into.